src/compiler/checker.ts

```diff
@@ -41050,7 +41050,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             forEach(node.name.elements, checkSourceElement);
         }
         // For a parameter declaration with an initializer, error and exit if the containing function doesn't have a body
-        if (isParameter(node) && node.initializer && nodeIsMissing((getContainingFunction(node) as FunctionLikeDeclaration).body)) {
+        if (hasInitializer(node) && isParameterDeclaration(node) && nodeIsMissing((getContainingFunction(node) as FunctionLikeDeclaration).body)) {

```

Comment:
Why use `hasInitializer`? `initializer` is a valid property for all of the types `node` can be here. The `hasInitializer` function potentially has poor performance as it performs an unconditional read of `initializer` for any `Node`. If `hasInitializer` doesn't get inlined, it could end up with a megamorphic IC based on how often it is called from other locations.

```suggestion
        if (node.initializer && isParameterDeclaration(node) && nodeIsMissing((getContainingFunction(node) as FunctionLikeDeclaration).body)) {
```

Reply:
Oke. Thanks for the detailed feedback.